File: solutions/leetcode_647.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        result = 0
        for i in range(len(s)):
            for j in range(i, len(s)):
                if s[i] != s[j]:
                    continue
                sub = s[i:j + 1]
                if sub == sub[::-1]:
                    result += 1
        return result


if __name__ == '__main__':
    given_s = "aaa"
    expected = 6

    solution = Solution()
    result = solution.countSubstrings(given_s)
    assert result == expected

File: solutions/leetcode_647.swift
class Solution {
    func helper(_ l:Int, _ r:Int, _ chars:[Character]) -> Int {
        var result = 0
        var l = l
        var r = r
        while l >= 0 && r < chars.count && chars[l] == chars[r] {
            result += 1
            l -= 1
            r += 1
        }
        return result
    }

    func countSubstrings(_ s: String) -> Int {
        var result = 0
        let chars = Array(s)
        for i in 0..<s.count {
            result += helper(i, i, chars)
            result += helper(i, i + 1, chars)
        }
        return result
    }
}
